libr/bin/bin.c

```diff
@@ -300,102 +328,109 @@ R_API bool r_bin_load_io(RBin *bin, int fd, ut64 baseaddr, ut64 loadaddr, int xt
 	RBinFile *binfile = NULL;
 	int tfd = -1;
 
-	r_return_val_if_fail (bin && io && fd >= 0 && (st64)sz >= 0, false);
+	bool is_debugger = iob->fd_is_dbg (io, opt->fd);
+	const char *fname = iob->fd_get_name (io, opt->fd);
 
-	bool is_debugger = iob->fd_is_dbg (io, fd);
-	const char *fname = iob->fd_get_name (io, fd);
-	if (loadaddr == UT64_MAX) {
-		loadaddr = 0;
+	bin->rawstr = opt->rawstr;
```

Comment:
this was done before in r_bin_load and I moved it to r_bin_open_io to be sure it's always set, but... I'll need to fix this in the future because it doesn't make sense to change the RBin rawstr property because you open a new BinFile.